### DISCUSSION TITLE
Note-by-note adjustments in realtime

### DISCUSSION THREAD

**Discussion by ghost** [FluidSynth/fluidsynth#972] [General]
I use jOrgan with Fluidsynth. It would be useful to adjust note-by-note volume, pitch, filter at least. Is it possible to do it with the correct midi messages using jOrgan or would it take a lot of further development in Fluidsynth? Polyphone and Viena are good but not realtime. Can it even be done with Fluidsynth?

**Comment by derselbst** [comment]
Automation can be achieved by sending MIDI control messages.

* For volume, send CC11 or CC7, or use note-on velocity.
* For pitch, send MIDI Pitch events.
* For filter (I guess you mean the low pass filter cutoff), you must define modulators in the soundfont you use. The modulator defines how you want the filter to act. You can freely choose any MIDI message for this purpose.

Also refer to [the wiki](https://github.com/FluidSynth/fluidsynth/wiki/FluidFeatures).

**Reply by ghost** [reply]
Many thanks for your reply. I currently have volume controls for each preset but have not been able to figure out midi messages to control pitch and volume for each note. As I see it now, note-on velocity may be the only answer.
